### ISSUE TITLE
clean installation erros

### ISSUE THREAD

**Issue by malkavi** [nazgul26/PHPRecipebook#10]
I have installed PHPRecipebook with a new db in my Synology NAS.
After fixing core_ingredients.sql, I still have errors. I have added ingredients and a new recipe, but I can't view it. 

```
Error: Table reviews for model Review was not found in datasource default.
```

And the "View" button when you have saved the recipe, into the edit mode, make a call to "PHPRecipebook/Recipes/view" without the recipe id. -> "PHPRecipebook/Recipes/view/1"

PD: I have tried to update from 4.0.9 with the same result, that's the reason to try from scratch.

**Comment by russbauer** [comment]
I am seeing the same error here and am just starting to dig into it. If you've come across any answers since posting this, I'd appreciate your thoughts.

I am also working with a clean install, but am on Linux Economy hosting via Godaddy.

If you'd like to see it for yourself, go to https://www.bauerhome.org/phprecipebook/ then click on Recipes, then click "View" on the only recipe you see. The result will be the same error malkavi is seeing, which seems to have a relationship to only the viewing of a recipe.

Here is the stack trace:

Missing Database Table
Error: Table reviews for model Review was not found in datasource default.

Notice: If you want to customize this error message, create phprecipebook/View/Errors/missing_table.ctp

Stack Trace
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php line 3644 → Model->setSource(string)
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php line 1387 → Model->getDataSource()
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Datasource/DboSource.php line 2516 → Model->schema()
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Datasource/DboSource.php line 1757 → DboSource->fields(Review, string, string)
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Datasource/DboSource.php line 1257 → DboSource->generateAssociationQuery(Recipe, Review, string, string, array, array, boolean)
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Datasource/DboSource.php line 1137 → DboSource->queryAssociation(Recipe, Review, string, string, array, array, boolean, array, integer, array)
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php line 3022 → DboSource->read(Recipe, array)
APP/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php line 2994 → Model->_readDataSource(string, array)
APP/Controller/RecipesController.php line 149 → Model->find(string, array)
[internal function] → RecipesController->view(string)
APP/Vendor/cakephp/cakephp/lib/Cake/Controller/Controller.php line 490 → ReflectionMethod->invokeArgs(RecipesController, array)
APP/Vendor/cakephp/cakephp/lib/Cake/Routing/Dispatcher.php line 193 → Controller->invokeAction(CakeRequest)
APP/Vendor/cakephp/cakephp/lib/Cake/Routing/Dispatcher.php line 167 → Dispatcher->_invoke(RecipesController, CakeRequest)
APP/webroot/index.php line 113 → Dispatcher->dispatch(CakeRequest, CakeResponse)

**Comment by russbauer** [comment]
Unfortunately this isn't as easy as adding table reviews with field Review. I tried adding the table reviews, with field Review then updated the schema by doing a ./Console/cake schema update, then I removed all files from ./tmp/cache/models and persistent directories and I'm getting new SQL messages about missing fields within the reviews table.

I wonder if somewhere along the line I missed getting the reviews table in one of the SQL files, or if maybe it was thought to be there initially via upgrade from a previous version(?).

If someone can provide me with a general idea of what the reviews table 'might' look like, I'd be willing to do some testing.

**Comment by nazgul26** [comment]
Something must have gone wrong with the cake schema update.  I have not done a clean install test after I added the reviews.  I did apply the review tables to a previous install though.  I can't dig into this now though, wish I could.  So please hold tight and I will get this fixed as soon as I can.

**Comment by nazgul26** [comment]
The MySQL version of the schema is still maintained in 
Config\SQL\schema_mysql.sql

The cake schema definition is used to extract to other db structures (i.e. Postgresql).

So the review table is:

CREATE TABLE reviews (
    id INT NOT NULL AUTO_INCREMENT,
    recipe_id INT NOT NULL REFERENCES recipes(id) ON DELETE CASCADE,
    comments VARCHAR(255) NOT NULL,
    created DATETIME,
    rating INT,
    user_id INT NULL REFERENCES users(id) ON DELETE SET DEFAULT ON UPDATE CASCADE,
    PRIMARY KEY (id),
    UNIQUE KEY (recipe_id,comments,user_id)
);

**Comment by russbauer** [comment]
Creating the review table, as specified above by nazgul26 seems to have done the trick. It did not work though until I emptied the tmp items. I'm including all steps I completed to get me on a functional install. Hopefully this helps others.
1. Create the review table by the following:
   CREATE TABLE reviews (
   id INT NOT NULL AUTO_INCREMENT,
   recipe_id INT NOT NULL REFERENCES recipes(id) ON DELETE CASCADE,
   comments VARCHAR(255) NOT NULL,
   created DATETIME,
   rating INT,
   user_id INT NULL REFERENCES users(id) ON DELETE SET DEFAULT ON UPDATE CASCADE,
   PRIMARY KEY (id),
   UNIQUE KEY (recipe_id,comments,user_id)
   );
2. Ran: ./Console/cake schema update  (although I'm not convinced that this step was necessary) and when complete I was still getting SQL errors when viewing Recipes.
3. When I removed all files from ./tmp/cache/models and persistent directories it seems to have done the trick.

Thanks for your assistance with this nazgul26. I think I'm good now, and suspect this will help malkavi too.

Cheers!
- Russ
